numba_dpex/dpnp_iface/arrayobj.py

```diff
@@ -126,19 +126,20 @@ def _parse_device_filter_string(device):
     elif isinstance(device, str):
         return device
     else:
-        raise TypeError(
-            "The parameter 'device' is neither of "
-            + "'str' nor 'types.StringLiteral'"
-        )
+        return None
```

Comment:
A `TypeError` should be raised if the device is not a string. Silently returning `None` can hide downstream issues.

Reply:
Fixed.